src/emmy/value.cljc

```diff
@@ -438,6 +439,55 @@
        (exact? [_] true)
        (kind [_] goog.math.Long))))
 
+#?(:cljs
+  ;; We find it convenient to be able to decorate "vanilla" JavaScript
+  ;; functions with metadata. In Clojurescript, using `with-meta` on a
+  ;; native function will replace it with an AFn (abstract function)
+  ;; instance, basically an object holding the function and its metadata
+  ;; in two ordinary properties. Such an object does not have the Function
+  ;; prototype, and is therefore not directly callable inJS.
+  ;;
+  ;; To fix this, we allow native functions to store metadata
+  ;; in a symbol-indexed property. The Clojurescript implementation
+  ;; is arguably more "hygienic," and gets around the difficulty of
+  ;; cloning the function as the `IWithMeta` contract would require,
+  ;; but this implementation should be safe enough for our purposes.
+  ;;
+  ;; By using a symbol instead of a string as the key for the metadata,
+  ;; the metadata annotation will not be visible among the object's other
+  ;; normal properties.
+   (let [metadata-symbol (. js/Symbol for "Symbol.__emmy_meta__")]
+     (extend-type function
+       IMeta
+       (-meta [f] (or (gobject/get f metadata-symbol) {})))
```

Comment:
I think cljs code prefers "nil" as the no-metadata-attached case... what do you think?